src/libsyntax/errors/mod.rs

```diff
@@ -38,10 +38,6 @@ pub enum RenderSpan {
     /// of hypothetical source code, where each `String` is spliced
     /// into the lines in place of the code covered by each span.
     Suggestion(CodeSuggestion),
```

Comment:
@nrc 

> Can we just get rid of RenderSpan? Suggestion seems different enough that it can just be a standalone struct at this point. And RenderSpan just bothers me.

I agree I'd like to get rid of `RenderSpan`, or maybe merge it with `MultiSpan`. I settled on the current setup as a kind of compromise. I still don't think we've quite got this stuff setup right. Maybe it'd be better to pass along both a `MultiSpan` and an `Option<CodeSuggestion>` instead of a `RenderSpan`.

Basically it seems to me that a message has the following parts:

- the message text
- the "primary spans" where it occurs (this is semantic info)

and then one of the following:

- a set of labels that should be used when rendering the message
- or, a set of span/replacement text to use as a suggestion

So when we render we either show the suggestion or, if there is no suggestion, then we show a snippet, with the given labels inserted (and other primary spans just underlined). 

This is *sort of* how the code works, but not quite as aligned as I'd like. I toyed with the idea of putting a bool into `MultiSpan` to indicate whether the labels were *labels* or *suggested replacement text*. This seems better than what we have -- and it merges `MultiSpan` and `RenderSpan` -- but `bool` always feels uncool. But then I decided to cut my losses and do what you see here, where things are desugared to a `RenderSpan`, and come back to this.

But basically the ways I can see to model that structure above are:

- MultiSpan with a bool, as described above, or perhaps an `enum` that chooses between a set of span+labels and a set of span+suggestions.
- An enum like RenderSpan that chooses between a `Snippet` (today's `MultiSpan`) and a `CodeSuggestion`. `CodeSuggestion` would not embed a `MultiSpan` but just a `Vec<(Span, String)>` pairs.

I guess either one would be better.